ironfish-cli/src/utils/timer.ts

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { createRootLogger, Logger, RawTransaction, TimeUtils } from '@ironfish/sdk'
+import { CliUx } from '@oclif/core'
+import { ProgressBar } from '../types'
+
+export class TransactionTimer {
+  private logger: Logger
+  private progressBar: ProgressBar | undefined
+  private startTime: number | undefined
+  private estimateInMs: number
+  private timer: NodeJS.Timer | undefined
+
+  constructor(spendPostTime: number, raw: RawTransaction, logger?: Logger) {
+    this.logger = logger ?? createRootLogger()
+    this.estimateInMs = Math.max(Math.ceil(spendPostTime * raw.spends.length), 1000)
+  }
+
+  displayEstimate() {
+    this.logger.log(
```

Comment:
you should just have a render function and the consumer should decide how to consume this. IE: commands should not use the logger, they should use `this.log` which is a base function in the Command.

Reply:
Thanks for the feedback, I was debating that too. Made the change